src/Rings/MPolyQuo.jl

```diff
@@ -725,14 +725,7 @@ MPolyQuo{MPolyElem_dec{fmpq, fmpq_mpoly}}
 """
 function quo(R::MPolyRing, I::MPolyIdeal) 
   q = MPolyQuo(R, I)
-  function im(a::MPolyElem)
-    parent(a) !== R && error("Element not in the domain of the map")
-    return MPolyQuoElem(a, q)
-  end
-  function pr(a::MPolyQuoElem)
-    return a.f
-  end
-  return q, MapFromFunc(im, pr, R, q)
+  return q, hom(R, q, gens(q))
```

Comment:
Now that I look at it again, I am not sure whether this was a good idea. Does the call to `hom` take into account that we only have to map variables to variables? 

@thofma : Do you think we should go back to the old version?

Reply:
What is the motivation for changing it (except to make clean it up)?

Reply:
I think I wanted the preimage functionality so that I can take lifts. But that has already been there, so... I don't remember, really. 

I just changed it back, but didn't push it yet. First I run the doctests again.